src/main/java/spoon/support/visitor/MethodTypingContext.java

```diff
@@ -267,6 +268,25 @@ private boolean isSameMethodFormalTypeParameter(CtTypeParameter scopeParam, CtTy
 				if (scopeBoundWildcard.getBoundingType().equals(scopeMethod.getFactory().Type().getDefaultBoundingType())) {
 					return true;
 				}
```

Comment:
Shouldn't the check above (line 241, idxOfScopeBoundTypeParam...) handle this already? The comment seems to indicate that (but it does not work for this case, though).

Reply:
ok - I had a quite similar solution but trapped then in the "adaptType" issue, which comes later, when the StackOverflow is fixed.